lib/register.php

```diff
@@ -394,6 +394,33 @@ function gutenberg_content_has_blocks( $content ) {
 	return has_blocks( $content );
 }
 
+/**
+ * Determine whether a $post or a string contains a specific block type.
+ * This test optimizes for performance rather than strict accuracy, detecting
+ * the block type exists but not validating its structure.
+ * For strict accuracy, you should use the block parser on post content.
+ *
+ * @since 3.6.0
+ *
+ * @param string                  $block_type Full Block type to look for.
+ * @param int|string|WP_Post|null $post Optional. Post content, post ID, or post object. Defaults to global $post.
+ * @return bool Whether the post content contains the specified block.
+ */
+function has_block( $block_type, $post = null ) {
+	if ( ! is_string( $post ) ) {
+		$wp_post = get_post( $post );
+		if ( $wp_post instanceof WP_Post ) {
+			$post = $wp_post->post_content;
+		}
+	}
+
+	if ( ! has_blocks( $post ) ) {
```

Comment:
This check could probably moved to the top of the function. If there are no blocks, getting the post before hand seems superfluous.